Core/src/main/java/com/theprogrammingturkey/comz/game/Game.java

```diff
@@ -268,7 +254,20 @@ public Game(String name)
 	 */
 	public PlayerWeaponManager getPlayersWeapons(Player player)
 	{
-		return playersGuns.computeIfAbsent(player, PlayerWeaponManager::new);
+		for (Player pl : getPlayers()) {
+			if (pl.equals(player)) {
+				return gamePlayers.get(player).getWeaponManager();
+			}
+		}
+		return null;
+	}
+
+	public List<Player> getPlayers() {
+		List<Player> players = new ArrayList<>();
+		for (GamePlayer value : gamePlayers.values()) {
+			if (value.getState().equals(GamePlayer.PlayerState.IN_GAME)) players.add(value.getPlayer());
+		}
+		return players;
 	}
```

Comment:
```suggestion
	public List<Player> getPlayers() {
		List<Player> players = new ArrayList<>(gamePlayers.values());
		players.removeIf(v -> !v.getState().equals(GamePlayer.PlayerState.IN_GAME));
		return players;
	}
```

I don't know Java, but this looks about right for how it should be in most langs (the removeif call should work)
**DO NOT APPLY BLINDLY.** Test first.